stencil_benchmarks/benchmarks_collection/stencils/cuda_hip/mixin.py

```diff
@@ -114,7 +114,8 @@ def template_args(self):
                     dry_runs=self.dry_runs,
                     timers=self.timers,
                     strides=self.strides,
-                    index_type=self.index_type)
+                    index_type=self.index_type,
+                    implementation_name=self.template_file().partition('.')[0])
```

Comment:
> But if we for example run using two different domain sizes, template_file is the same, so this does not really help in general, or does it?

It's true that it only partly helps. IIRC one can infer the domain size by the grid size reported by NCU though?
When I was looking into this, distinguishing the kernel names was helpful enough since I could figure out which size was each report then. For different sizes one can also launch the benchmarks separately and create separate `ncu-rep` files that encode the domain sizes as well.
Tbh I don't think that a hash adds much value towards the desired purpose.

Reply:
Running just a single kernel at a time was what I always did. But then we don’t need any change from now because you either know exactly what you have run or otherwise you have a problem because of missing information, right?
So I am for either completely fixing the problem (if it is one) or completely ignoring it.

Reply:
If we run each kernel with separate commands then we can avoid those issues and simplify things. I was just using the `sbench-<arch>-collection` to run the benchmarks and collect results in batches.
It would still be useful to have a way to inspect and distinguish the generated code for each implementation by either encoding the implementation name in the function name or the filename saved to disk.